### PR TITLE
Workaround for PySide6 problem

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Workaround for problem from #5217. 

As PySide maintainers decide to monkeypatch enum module for python bellow 3.10 on import of PySide and napari is not only an application but also a library, and there is no possibility to guarantee that PySide will be the first imported library we cannot support using PySide6 > 6.3.1 on python < 3.10. 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
fail of test is connected with https://github.com/spyder-ide/qtpy/pull/374 and requires a new qtpy release

**Comment by psobolewskiPhD** [comment]
> fail of test is connected with [spyder-ide/qtpy#374](https://github.com/spyder-ide/qtpy/pull/374) and requires a new qtpy release

On macOS arm64 I see a vispy issue with pyside6 6.4.0 (but not 6.3.2, on python 3.10):
https://github.com/vispy/vispy/issues/2407
I wonder if that's also arm64 specific or pyside6 6.4.0 specific.

**Comment by Czaki** [comment]
> I wonder if that's also arm64 specific or pyside6 6.4.0 specific.

this is pyside 6.4.0 specyfic

**Review by jni — APPROVED** [review]

**Comment by andy-sweet** [comment]
@jni : thanks for the review! What are your thoughts about cherry picking this for v0.4.17?

@Czaki : would cherry picking this for v0.4.17 also require a new qtpy release?

**Comment by Czaki** [comment]
> @Czaki : would cherry picking this for v0.4.17 also require a new qtpy release?

no. This is required to be working with PySide 6.4.0. We do not ship bundles with PySide6, so it is not blocking for me. But python 3.10 with PySide6 will fail until the new qtpy release.

**Comment by jni** [comment]
> What are your thoughts about cherry picking this for v0.4.17?

I'm in favour! Thank you *SO MUCH* for all the work you have done for this release! 🙏 😭

File: napari/_qt/__init__.py [hunk 1]
```diff
@@ -27,6 +27,17 @@
         Path(PySide2.__file__).parent / 'Qt' / 'plugins'
     )
 
+if API_NAME == 'PySide6' and sys.version_info[:2] < (3, 10):
+    from packaging import version
+
+    if version.parse(QT_VERSION) > version.parse("6.3.1"):
+        raise RuntimeError(
```
Comment (andy-sweet) [inline napari/_qt/__init__.py:34]: I created an environment on my Intel mac with Python 3.9.13 and PySide6==6.4.0 (see [output of `pip list`](https://gist.github.com/andy-sweet/0269f3cfafad49a6fe0f41f8b4d4e984) for other details).

But when I run napari, I get the [enum error message related to the PySide6 issue](https://gist.github.com/andy-sweet/f7b3db105b233e00854726d37b04aa4c).

I was expecting to get this error first. Or at least an error from qtpy until a new version is released. Did I misunderstand the intent of this part of the PR?

Comment (andy-sweet) [reply napari/_qt/__init__.py:34]: I guess we're getting the enum-error as we try to show the original error?

**Comment by andy-sweet** [comment]
Definitely want to merge the change to `tox.ini` ASAP both for `main` and for `v0.4.17x` so that the pyside6 tests run and pass.

Based on my line comment on raising, I'm not sure about the rest of the PR, especially as something to cherry-pick for v0.4.17. But I may have misunderstood the bug case / proper setup here.

File: napari/_qt/__init__.py (hunk 1 above)
Comment (Czaki) [reply napari/_qt/__init__.py:34]: I just found out how many places we import qtpy directly...

Comment (Czaki) [reply napari/_qt/__init__.py:34]: Ok, I fix the code by proper move qt import. 

The PySide6 was imported in `magicgui` that means that our wold fixes in `napari._qt` may also not work.

Thanks for finding this before merging.

**Review by andy-sweet — APPROVED** [review]
After the latest commit, when I run `napari`, I get an error from `qtpy` related to [issue #374](https://github.com/spyder-ide/qtpy/pull/374) when I have pyside6 (6.4.0) and python 3.9, which is what I expect. After the next `qtpy` release, we should get the error added here.

I can avoid any errors by either upgrading python or downgrading pyside6, so this looks good to me now.

File: .github/workflows/test_pull_requests.yml [hunk 2]
```diff
@@ -78,6 +78,9 @@ jobs:
           - python: 3.9
             platform: ubuntu-latest
             backend: pyside6
+          - python: '3.10'
+            platform: ubuntu-latest
+            backend: pyside6
```
Comment (andy-sweet) [inline .github/workflows/test_pull_requests.yml]: I did just approve this, but I don't think we should add a workflow to CI that we know will fail until a new version of `qtpy` is released. Can we remove, disable, or comment this out with a TODO for now until that `qtpy` release is available?

**Comment by andy-sweet** [comment]
@Czaki : I'm planning to merge this soon and will cherry pick the commit to the v0.4.17x branch. I just have one remaining comment about the currently failing GitHub workflow.

File: .github/workflows/test_pull_requests.yml (hunk 2 above)
Comment (Czaki) [reply .github/workflows/test_pull_requests.yml]: commented

**Pull request merged**